src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -1439,6 +1449,42 @@ trait Namers extends MethodSynthesis {
       pluginsTypeSig(methSig, typer, ddef, resTpGiven)
     }
 
+    /**
+     * For every default argument, insert a method symbol computing that default
+     */
+    private def enterDefaultGetters(meth: Symbol, ddef: DefDef, vparamss: List[List[ValDef]], tparams: List[TypeDef]) {
+      val methOwner  = meth.owner
+      val search = DefaultGetterNamerSearch(context, meth, initCompanionModule = false)
+      var posCounter = 1
+
+      mforeach(vparamss){(vparam) =>
+        // true if the corresponding parameter of the base class has a default argument
+        if (vparam.mods.hasDefault) {
+          val name = nme.defaultGetterName(meth.name, posCounter)
+
+          search.createAndEnter { owner: Symbol =>
+            methOwner.resetFlag(INTERFACE) // there's a concrete member now
+            val default = owner.newMethodSymbol(name, vparam.pos, paramFlagsToDefaultGetter(meth.flags))
+            default.setPrivateWithin(meth.privateWithin)
+            default.referenced = meth
+            default.setInfo(ErrorType)
```

Comment:
This part is a bit gross, but we're not able to form the RHS of the default getter yet, so we can't assign the type completer.

Reply:
reminds me of our discussion on generalizing unit.synthetics to take tree creators rather than trees